### PR TITLE
 #3781 Allow drag&drop of layers/groups between groups

### LINKED ISSUE #3781: Allow drag&drop of layers/groups between groups
### Description
Currently drag&drop only reorders stuff inside the containing folder, moving layers/groups to different groups is not supported through drag & drop now: using the **group** combobox inside **Layer Settings** is the only way to accomplish this now.

What we need in this issue is to improve that behavior by implementing drag&drop of layers/groups to a different group (not only the possibility of reordering them in TOC).

### PR DESCRIPTION
**Pull request by vlt1**
## Description
Default group is not draggable anymore. When something is inserted before the Default group, the operation is canceled and a warning notification is shown.

## Issues
 - #3781 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
#3781 

**What is the new behavior?**
In the description

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Comment by vlt1** [comment]
What do you mean drag out of Default group? How that should work specifically?

**Comment by offtherailz** [comment]
I mean I can create a group at the same level of the default group, drag it inside the default group, but then I can not drag it out anymore ( I can not drag it on top of the default group, and this is correct, but I can not dragi it down too, and this is wrong)

**Comment by offtherailz** [comment]
When only default group is on the first level of the tree, there is no "below" to drag to, this is the reason of the bug I notified.

**Comment by offtherailz** [comment]
Another couple of bugs:
Peek 18-10-2019 10-57
I'm able to drag above default

**Comment by vlt1** [comment]
@offtherailz added dragout, please test

**Comment by vlt1** [comment]
> great job 👍
> My only fear is about unit tests. How can we ensure that these functionality will not break with some future changes? Can you add a test for presence of dummy stuff in certain circumstances?

I can't provide unit tests for this feature because current version of d&d library doesn't provide test utils. I think it could be broken only if you'll change some code in the d&d context creation or remove target groups.

File: web/client/components/TOC/fragments/GroupChildren.jsx [hunk 1]
```diff
@@ -13,8 +13,10 @@ require('./css/groupchildren.css');
 class GroupChildren extends React.Component {
     static propTypes = {
         node: PropTypes.object,
+        dndDummy: PropTypes.object,
```
Comment (offtherailz) [inline web/client/components/TOC/fragments/GroupChildren.jsx]: I think this can be removed

**Pull request merged**